scripts/cmake/config/default.cmake

```diff
@@ -783,21 +724,33 @@ add_config(
 
 if(BUILD_SHARED_LIBS)
     if(BUILD_EXTENDED_APIS)
-        set(BFM_VMM "eapis_shared")
+        set(DEFAULT_BFM_VMM "eapis_shared"
+            CACHE INTERNAL
+            "Default BFM VMM"
+        )
     else()
-        set(BFM_VMM "bfvmm_shared")
+        set(DEFAULT_BFM_VMM "bfvmm_shared"
+            CACHE INTERNAL
+            "Default BFM VMM"
+        )
     endif()
 else()
     if(BUILD_EXTENDED_APIS)
```

Comment:
BUILD_EXTENDED_APIS isn't a defined configuration anymore (which is correct, it shouldn't be). I think we should change this to just point to bfvmm_shared or bfvmm_static as defaults, and let people overide the default to eapis in their configuration file.